Assets/Scripts/Player/PlayerInput.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerInput: MonoBehaviour
+{
+    private PlayerMovement playerMovement;
+    public bool inputLocked;
+    public float manaPoints;
+    public int maxMana;
```

Comment:
I'm not sure these really belong with player input. Maybe make a separate player stats class that holds things like this, or put them in with the class in charge of casting spells.